ledger/apply/keyreg.go

```diff
@@ -77,9 +77,15 @@ func Keyreg(keyreg transactions.KeyregTxnFields, header transactions.Header, bal
 			}
 		}
 		record.Status = basics.Online
+		if params.Mining().Enabled {
+			record.LastHeartbeat = header.FirstValid
+		}
 		record.VoteFirstValid = keyreg.VoteFirst
 		record.VoteLastValid = keyreg.VoteLast
 		record.VoteKeyDilution = keyreg.VoteKeyDilution
+		if header.Fee.Raw >= params.Mining().GoOnlineFee && params.Mining().Enabled {
+			record.IncentiveEligible = true
+		}
```

Comment:
Interesting thing about this is that there's no way to prevent this fee from being used for other purposes as well (as far as I'm aware). The other purposes I can think of are covering the fees from other top level txns, and providing a fee credit for inner txns.

I wonder if it'd be worthwhile to somehow decrement the EvalParams fee credit by GoOnlineFee, and if there's not enough free credit to cover it, error?

I suppose there's not that much at stake here, since if GoOnlineFee is 2 Algos, that already greatly exceeds any other possible fee usage. Though if fees were to increase in the future, it'd be nice to get this right.

Reply:
True. A similar problem exists for extra fees for inner txns. _Maybe_ you added the extra fee in order to pay for congestion. But then you get to spend the surplus on inners to.

Reply:
Oh, but your point is stronger.  We _could_ handle this properly because it's in consensus. I'll think on it.